samza-core/src/main/java/org/apache/samza/util/ConfigUtil.java

```diff
@@ -67,4 +72,42 @@ public static Config applyRewriter(Config config, String rewriterName) {
     LOG.info("Re-writing config with {}", rewriter);
     return rewriter.rewrite(rewriterName, config);
   }
+
+  /**
+   * Load full job config with {@link ConfigLoaderFactory} when present.
+   *
+   * @param original config
+   * @return full job config
+   */
+  public static Config loadConfig(Config original) {
+    JobConfig jobConfig = new JobConfig(original);
+    Config fullConfig = original;
+
+    if (jobConfig.getConfigLoaderFactory().isPresent()) {
+      ConfigLoaderFactory factory = ReflectionUtil.getObj(jobConfig.getConfigLoaderFactory().get(), ConfigLoaderFactory.class);
+      ConfigLoader loader = factory.getLoader(original.subset(ConfigLoaderFactory.CONFIG_LOADER_PROPERTIES_PREFIX));
+      // overrides config loaded with original config, which may contain overridden values.
+      fullConfig = override(ConfigUtil.rewriteConfig(loader.getConfig()), original);
+    }
```

Comment:
Is it an exceptional case for ConfigLoaderFactory to not be present here?

Reply:
Eventually Yes, but during the migration, there will be cases where jobs are still following the legacy submission flow, but still goes to this new method in code.

For example, this util will be called by LocalApplicationRunner too, which will first support both legacy and new flow.

Reply:
The method name `loadConfig` is a bit misleading if it just might return the config that is passed in. If you still prefer to allow this to succeed when `ConfigLoaderFactory` is not present, then maybe at least rename the method to help clarify what it might do, so that someone doesn't accidentally call it in an unexpected way.

Reply:
Is there something you can do to clarify that this may not actually load any additional configs?

Reply:
Updated to disallow the case when ConfigLoaderFactory is not present.